pkg/reconciler/pipelinerun/resources/pipelinerunresolution_test.go

```diff
@@ -939,9 +940,39 @@ func TestIsSkipped(t *testing.T) {
 		expected: map[string]bool{
 			"mytask10": false,
 		},
+	}, {
+		name: "tasks-when-expressions-in-finally-passed",
```

Comment:
these tests can be moved to `TestResolvedPipelineRunTask_IsFinallySkipped`

Reply:
PR - https://github.com/tektoncd/pipeline/pull/3744 will not be needed if we change `IsFinallySkipped`.